### DISCUSSION TITLE
Undefined index when no listeners defined for event

### DISCUSSION THREAD

**Discussion by IvanChernikov** [livewire/livewire#3056] [7. Bugs (archived)]
https://github.com/livewire/livewire/blob/418eb119fa87b70b7c70e74e10c4bee28e8f740f/src/ComponentConcerns/ReceivesEvents.php#L70-L75

If no event listeners defined for a given event type, an `ErrorException("Undefined index: {$event}")` is raised. Should validate if any listeners exist for event

**Comment by FR6** [comment]
See PR #2427.

**Comment by calebporzio** [comment]
I agree there should be a better error for this case, and also livewire should support dynamic backend listeners.

Would love a PR for at least the first one (would be a simple one)
